attestation-agent/kbs_protocol/src/client/rcar_client.rs

```diff
@@ -104,21 +104,33 @@ impl KbsClient<Box<dyn EvidenceProvider>> {
         let request = Request {
             version: String::from(KBS_PROTOCOL_VERSION),
             tee,
-            extra_params: String::new(),
+            extra_params: serde_json::Value::String(String::new()),
         };
 
         debug!("send auth request to {auth_endpoint}");
 
-        let challenge = self
+        let resp = self
             .http_client
             .post(auth_endpoint)
             .header("Content-Type", "application/json")
             .json(&request)
             .send()
-            .await?
-            .json::<Challenge>()
             .await?;
 
+        match resp.status() {
+            reqwest::StatusCode::UNAUTHORIZED => {
+                let error_info = resp.json::<ErrorInformation>().await?;
+                bail!(
+                    "KBS request unauthorized, ErrorInformation: {:?}",
+                    error_info
+                );
+            }
+            _ => {
+                debug!("KBS request OK");
```

Comment:
Maybe we should explicitly check for the success status code here and use the fall through case to catch other types of errors.

Reply:
fixed (the same way it's done for the `attest_response` below).